prompt/prompt-model/src/commonMain/kotlin/ai/koog/prompt/message/Message.kt

```diff
@@ -73,6 +73,27 @@ public sealed interface Message {
         Tool
     }
 
+    /**
+     * Represents a message that includes one or more attachments.
+     *
+     * This interface extends the [Message] interface, adding support for managing
+     * additional content in the form of attachments. Attachments can be of various
+     * types such as images, videos, audio files, or other file formats. Implementing
+     * classes should define specific details of the message alongside its attachments.
+     *
+     * @property attachments A list of attachments associated with the message.
+     * Each attachment contains specific content and metadata.
+     */
+    @Serializable
+    public sealed interface MessageWithAttachments : Message {
```

Comment:
nit: `With` doesn't look nice in the interface name. And `Message` also feels redundant, since it's inside the `Message` class already. How about simply `Attachments`? Looks like it aligns better with existing naming convention, like `Response` or `Tool` interfaces